apps/scan/frontend/src/components/audio_diagnostic_modal_button.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { DiagnosticOutcome } from '@votingworks/types';
+import { Button, Icons, Modal, P } from '@votingworks/ui';
+import { useSound } from '../utils/use_sound';
+import * as api from '../api';
+
+export interface AudioDiagnosticModalProps {
+  setVisible: (visible: boolean) => void;
+}
+
+export function AudioDiagnosticModal(
+  props: AudioDiagnosticModalProps
+): JSX.Element {
+  const { setVisible } = props;
+  const playSound = useSound('success');
+  React.useEffect(playSound, [playSound]);
+
+  const { isLoading, mutate: logOutcome } =
+    api.logAudioDiagnosticOutcome.useMutation();
+
+  const onConfirm = React.useCallback(
+    (outcome: DiagnosticOutcome) => {
+      logOutcome({ outcome });
+      setVisible(false);
+    },
+    [logOutcome, setVisible]
+  );
```

Comment:
IMO it's not necessary to `useCallback` on these tiny functions because it's unclear if there's a performance gain and it sacrifices some readability.

Reply:
Yeah, force of habit. Also hadn't considered that it might be a negative for readability, since I've gotten so used to it, so that's good perspective - will update.
Related to the previous comment on consistency, though - there could be some benefit to making the default pattern the one that adds cumulative value (we've already seen perf hits from lots of unnecessary re-renders when stuff changes at the top of the tree, which is more of a death-by-a-thousand-cuts sort of thing)